javascript/operators/import_meta.json

```
@@ -45,6 +45,48 @@
             "standard_track": true,
             "deprecated": false
           }
+        },
+        "resolve": {
+          "__compat": {
+            "description": "<code>import.meta.resolve</code>",
+            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Operators/import.meta.resolve",
+            "spec_url": "https://html.spec.whatwg.org/multipage/webappapis.html#hostgetimportmetaproperties",
+            "support": {
+              "chrome": {
+                "version_added": "105"
+              },
+              "chrome_android": "mirror",
+              "deno": {
+                "version_added": "1.24"
+              },
+              "edge": "mirror",
+              "firefox": {
+                "version_added": "106"
+              },
+              "firefox_android": "mirror",
+              "ie": {
+                "version_added": false
+              },
+              "nodejs": {
+                "version_added": false,
+                "notes": "Node.js includes an implementation of import.meta.resolve() when using a flag (--experimental-import-meta-resolve), but this implementation returns a Promise and is therefore incompatible with browser behaviour."
```

Rather than marking this as unsupported and adding a note explaining how it is supported in its own way, we should set the version number, set `partial_implementation` to `true`, add `flags` and have a simpler note stating how its implementation differs from normal:

```suggestion
                "version_added": "???",
                "partial_implementation": true,
                "flags": [
                  {
                    "type": "runtime_flag",
                    "name": "--experimental-import-meta-resolve"
                  }
                ],
                "notes": "In Node.js, this method returns a Promise."
```

---

```suggestion
                "version_added": "???",
                "partial_implementation": true,
                "flags": [
                  {
                    "type": "runtime_flag",
                    "name": "--experimental-import-meta-resolve"
                  }
                ],
                "notes": "Returns a Promise."
```

---

Looks like the exact version was 12.6.2:

https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V12.md#2020-04-08-version-12162-erbium-lts-codebytere

Should that be listed as 12.6?

---

Thanks for tracking that down -- yes, the version number should be set to "12.6"!

---

Exciting:

```
version_added: "12.6" is NOT a valid version number for nodejs
    Valid nodejs versions are: 0.10.0, 0.12.0, 4.0.0, 5.0.0, 6.0.0, 6.5.0, 7.0.0, 7.5.0, 7.6.0, 7.7.0, 7.10.0, 8.0.0, 8.3.0, 8.5.0, 8.7.0, 8.10.0, 9.3.0, 10.0.0, 10.4.0, 10.5.0, 10.7.0, 10.9.0, 11.0.0, 11.7.0, 12.0.0, 12.5.0, 12.9.0, 12.11.0, 12.17.0, 13.0.0, 13.2.0, 14.0.0, 14.3.0, 14.5.0, 14.6.0, 14.7.0, 14.8.0, 15.0.0, 15.4.0, 15.7.0, 15.12.0, 16.0.0, 16.1.0, 16.4.0, 16.5.0, 16.6.0, 16.7.0, 16.9.0, 16.11.0, 16.14.0, 16.15.0, 16.17.0, 17.0.0, 17.2.0, 17.3.0, 17.5.0, 18.0.0, 19.0.0
```

Does that mean I should add the version to https://github.com/lgarron/browser-compat-data/blob/4f527a4af1dc3b92dfb42bec1e3d0b565934aca9/browsers/nodejs.json#L191 ?

Would there be an issue with adding 12.6.2 instead, since 12.6.0 is technically inaccurate?

---

To be honest, I don't know why we append all NodeJS version numbers with ".0" when we [don't track patch versions in BCD](https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#choosing-a-version-number).  I'm going to open an issue suggesting the removal of the patch version.  If you could add "12.6.0" to the browser data, that would be great!